sos/report/plugins/mellanox_firmware.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (C) 2023 Nvidia Corporation, Alin Serdean <aserdean@nvidia.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin
+import os
+import time
+
+
+class MellanoxFirmware(Plugin, IndependentPlugin):
+
+    short_desc = 'Nvidia(Mellanox) firmware tools output'
+
+    plugin_name = "mellanox_firmware"
+    profiles = ('hardware', 'system')
+    packages = ('mst', 'mstflint')
+
+    MLNX_STRING = "Mellanox Technologies"
+
+    def check_enabled(self):
+        """
+        Checks if this plugin should be executed at all.
+        We will only enable the plugin if there is a
+        Mellanox Technologies network adapter
+        """
+        lspci = self.exec_cmd("lspci -D -d 15b3::0200")
+        if lspci['status'] != 0:
+            return False
+        return self.MLNX_STRING in lspci['output']
+
+    def collect(self):
```

Comment:
It was suggested here: https://github.com/sosreport/sos/pull/3407#discussion_r1396383691

Reply:
Yeah, since we're modifying system state, I think it best if we do this during `collect()`, and that requires using `collect_cmd_output()` for the just-in-time writing to the plugin dir.

Reply:
Ah I see, it makes sense (and I should read whole PR discussion before reviewing).

Maybe then it is worth having a comment abou the unusual approach in the code?